.github/workflows/haskell.yml

```
@@ -62,22 +62,18 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        ghc: ["8.10.7", "9.2.8", "9.6.6", "9.8.2", "9.10.1", "9.12.1"]
+        ghc: ["9.6", "9.8", "9.10", "9.12"]
```

We need to pin ghc versions exactly, it is not a good idea to allow CI to change versions from under you at an arbitrary point. We've always done it that way, so why did you decide to change it all of a sudden?
```suggestion
        ghc: ["9.6.7", "9.8.4", "9.10.2", "9.12.2"]
```

---

I actually do not agree.

GHC version `9.12.2` is only supposed to contain bug fixes from `ghc-9.12.1` . I, in the very UNLIKLEY event a but fix version does cause a regression, CI is the perfect place to find out about it.

---

@erikd I appreciate your opinion, but CI suppose to be reproducible as much as possible.
That is why we use `index-state` and pin other dependencies.

> In the very UNLIKLEY event a bug fix version does cause a regression, CI is the perfect place to find out about it.

In this unlikely scenario we will still catch it, but only after manual update of ghc version on CI, which we usually update often.

I'd rather regressions like this be caught on a PR that updates the ghc version, then on a PR that changes the actual code in one of the packages in the repo.

I understand that every team has their own approach, but in cardano-ledger and in cardano-base this is the approach we take, so I'd appreciate if it was not changed because you are of a different opinion.